indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -149,6 +151,10 @@ public String getClasspathPrefix()
   @Override
   public TaskStatus run(TaskToolbox toolbox) throws Exception
   {
+    //Setting up UsedSegmentLister which is used by DatasourcePathSpec to get the list of used
+    //segments
+    DatasourcePathSpec.setUsedSegmentLister(new OverlordActionBasedUsedSegmentLister(toolbox));
```

Comment:
Thanks @himanshug. So it sounds like there isn't much of a solution other that continuing to move towards having the overlord be in every druid deployment. That's okay, I think, since I think we want to do that anyway.